presto-main/src/main/java/com/facebook/presto/sql/analyzer/Analyzer.java

```diff
@@ -83,12 +82,12 @@ static void verifyNoAggregateWindowOrGroupingFunctions(FunctionRegistry function
         List<FunctionCall> windowExpressions = ExpressionTreeUtils.extractExpressionsOfTypeUsingPredicate(
                 ImmutableList.of(predicate),
                 FunctionCall.class,
-                WINDOW_EXTRACTOR_FUNCTION);
+                ExpressionTreeUtils::isWindowFunction);
 
         List<GroupingOperation> groupingFunctions = ExpressionTreeUtils.extractExpressionsOfTypeUsingPredicate(
```

Comment:
I would make function version of `extractExpressionsOfTypeUsingPredicate` without predicate.